src/main/java/frc/robot/subsystems/swerve/swervestatehelpers/SwerveStateHelper.java

```diff
@@ -0,0 +1,115 @@
+package frc.robot.subsystems.swerve.swervestatehelpers;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import frc.robot.constants.MathConstants;
+import frc.robot.subsystems.swerve.Swerve;
+import frc.robot.subsystems.swerve.SwerveConstants;
+import frc.robot.subsystems.swerve.SwerveMath;
+import frc.robot.subsystems.swerve.modules.ModuleUtils;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static frc.robot.subsystems.swerve.swervestatehelpers.AimAssistUtils.getRotationAssistedSpeeds;
+import static frc.robot.subsystems.swerve.swervestatehelpers.RotateAxis.BACK_LEFT_MODULE;
+import static frc.robot.subsystems.swerve.swervestatehelpers.RotateAxis.BACK_RIGHT_MODULE;
+import static frc.robot.subsystems.swerve.swervestatehelpers.RotateAxis.FRONT_LEFT_MODULE;
+import static frc.robot.subsystems.swerve.swervestatehelpers.RotateAxis.FRONT_RIGHT_MODULE;
+
+public class SwerveStateHelper {
+
+    private final Supplier<Pose2d> robotPoseSupplier;
+    private final Swerve swerve;
+    private final SwerveConstants swerveConstants;
+    private final Supplier<Optional<Translation2d>> noteTranslationSupplier;
+
+    public SwerveStateHelper(Supplier<Pose2d> robotPoseSupplier, Supplier<Optional<Translation2d>> noteTranslationSupplier,
+            Swerve swerve) {
+        this.robotPoseSupplier = robotPoseSupplier;
+        this.swerve = swerve;
+        this.swerveConstants = swerve.getConstants();
+        this.noteTranslationSupplier = noteTranslationSupplier;
+    }
+
+    public ChassisSpeeds applyAimAssistOnInputsSpeeds(AimAssist aimAssistState, ChassisSpeeds inputSpeeds) {
```

Comment:
```suggestion
    public ChassisSpeeds applyAimAssistOnSpeeds(AimAssist aimAssistState, ChassisSpeeds speeds) {
```
speeds or chassisSpeeds